packages/ERTP/src/mathHelpers/natMathHelpers.js

```diff
@@ -18,15 +18,22 @@ const identity = 0;
  *
  * @type {MathHelpers}
  */
-const natMathHelpers = harden({
+const natMathHelpers = {
   doCoerce: Nat,
   doGetEmpty: _ => identity,
   doIsEmpty: nat => nat === identity,
   doIsGTE: (left, right) => left >= right,
   doIsEqual: (left, right) => left === right,
   doAdd: (left, right) => Nat(left + right),
   doSubtract: (left, right) => Nat(left - right),
-});
+  doFind: (left, searchParameters) => {
+    if (natMathHelpers.doIsGTE(left, searchParameters)) {
+      return searchParameters;
```

Comment:
This is frugal. As used by `eventualWithdraw` you're withdrawing the least amount of rights that satisfies the pattern. This is opposite of the set variants, which leaves me confused about what the abstract meaning of the operation is.